### PR TITLE
Feat/integration tests

### PR DESCRIPTION
**Pull request by shkurata**

### CODE REVIEW

File: api/test/competency-mutations.spec.ts [hunk 1]
```diff
@@ -10,6 +10,28 @@ import {
...
+      }
+    `,
+    {},
+  );
+  expect(response).not.toHaveProperty('errors');
+  const [competence1, competence2] = response.data.allRootCompetencies;
+  expect(competence1?.id).toBe(competenceId1);
+  expect(competence2?.id).toBe(competenceId2);
+}
+
```
Comment (Moeriki) [inline api/test/competency-mutations.spec.ts:34]: You can write helpers for testing but don't do assertions (`expect`) in a helper. This obfuscates what a test is actually testing. Developer shouldn't have to click around for that.

DRY is not so important for tests. Tests should very clearly contain all steps.

We could gather some common queries, or make helpers to fetch certain types of data. But I would honestly not even do that.

File: api/test/competency-queries.spec.ts [hunk 2]
```diff
@@ -6,6 +6,28 @@ import { prisma } from '../src/prisma';
...
+    `,
+    {
+      spec: { locale },
+      variables: { id },
+    },
+  );
+  expect(response).not.toHaveProperty('errors');
+  expect(response).toHaveProperty('data.competency.title', title);
+}
+
```
Comment (Moeriki) [inline api/test/competency-queries.spec.ts:30]: Why was this refactored?

**Review by Moeriki — CHANGES_REQUESTED** [review]

File: api/test/competency-queries.spec.ts (hunk 2 above)
Comment (shkurata) [reply api/test/competency-queries.spec.ts:30]: I first wanted to reuse the query to fetch the competence by id and I made a function that can be exported and then I saw that the same query was used in almost all tests, so I replaced the queries with the function. Then I realised that this query will not be of use in my tests, but I left the function to be reviewed.

Comment (Moeriki) [reply api/test/competency-queries.spec.ts:30]: Ok. The same remarks as the other file apply here.

**Review by Moeriki — APPROVED** [review]

**Review by SiteSeerBE — APPROVED** [review]

**Pull request merged**